File: python3/test/doubles/InMemoryProductCatalog.py
from src.repository.ProductCatalog import ProductCatalog


class InMemoryProductCatalog(ProductCatalog):
    def __init__(self, products: list):
        self.products = products

    def get_by_name(self, name: str):
        return next(filter(lambda product: product.get_name() is name, self.products), None)

File: typescript/test/doubles/InMemoryProductCatalog.ts
import Product from '../../src/domain/Product';
import { ProductCatalog } from '../../src/repository/ProductCatalog';

class InMemoryProductCatalog implements ProductCatalog {
  private products: Product[];

  public constructor(products: Product[]) {
    this.products = products;
  }

  public getByName(name: string): Product {
    return this.products.find(p => p.getName() === name);
  }
}

export default InMemoryProductCatalog;

